app/models/event.rb

```diff
@@ -7,7 +7,10 @@ class Event < ActiveRecord::Base
 
   belongs_to :venue, class_name: 'Sponsor'
   has_many :sponsorships
-  has_many :sponsors, through: :sponsorships
+  has_many :sponsors, -> { where('sponsorships.level' => nil) }, through: :sponsorships, source: :sponsor
```

Comment:
Is there any value in keeping 'standard' sponsors as I see this causing confusion moving forward.
Adding a migration moving all the existing sponsorships to the lowest available level would enable us to display sponsors in past events and not keep functionality that we don't want to be utilised any longer.

Reply:
In my first version of this PR the new column in the connection table had the default to `bronze`.
I kept the standard level of sponsorship for retro compatibility and to allow future events without levels but if all the future events will have them I will simply remove the relationship and reinstate the default value if instead, the problem is more the cleanness of the form we could add a switch to hide the standard or levelled input.

Reply:
Can we then split this up so that
1. In the form either a 'standard' sponsor can be set, or 'gold, bronze or silver' - **add some information text** letting admin/organisers know that they should either select `standard' sponsors or leveled sponsors
2. Condition the views so that either standard sponsors are rendered (displayed a level here would also be unecessary so drop the standard title in the member view, maybe keep it in the admin view) or different sponsorship tiers and the relevant sponsors

It would have been nice to have this association to be implemented [using nested attributes](https://www.pluralsight.com/guides/ruby-on-rails-nested-attributes) which would simplify the views as well as the relationships and model associations, but it would be a bit of work and I think we need this in ASAP so it's maybe something to consider for future optimisation of this functionality.

Reply:
that makes sense, I tried to keep the view as similar as it is currently but yes, with nested attributes could make everything cleaner.

for the new sponsor section, I'm thinking to replace the four dropdowns with: 
on top a checkbox to select levelled or standard sponsors then under it a three columns grid with a line for each sponsor and on the first column a dropdown menu for the sponsors and a series of three radio button for the level if selected or hide it if not needed then a button to remove the sponsors, and after the grid a button to add a new sponsor.

Because this functionality is needed immediately I think we could merge this PR as it is and I'll open a new one for the new form.

Reply:
@jabawack81 ok but can we add some information text informing editors that they should either select standard or any of the other levels so that regardless of any future work being done it is clear why those four levels are set.

Also can you please rebase! Thanks.

Reply:
done.

![Screenshot_2020-08-11_11-24-33](https://user-images.githubusercontent.com/432450/89886931-5cc9f100-dbc5-11ea-8a32-68396e927595.png)

Sorry I resolved the conversation by mistake.

Reply:
No worries. Thanks for making the change.